plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/internal/EclipseBasedN4JSWorkspace.java

```diff
@@ -226,7 +227,7 @@ public boolean visit(IResource resource) throws CoreException {
 				return Iterators.unmodifiableIterator(result.iterator());
 			}
 		}
-		return Iterators.emptyIterator();
+		return Iterators.unmodifiableIterator(Collections.emptyIterator());
```

Comment:
why add `Iterators.unmodifiableIterator(`? Isn't the empty iterator unmodifiable anyway?

Reply:
Nope. empty iterator is NOT unmodifiable :(

Reply:
done